cmd/metal-api/internal/datastore/integer.go

```diff
@@ -26,16 +26,24 @@ type Integerinfo struct {
 
 // initIntegerPool initializes a pool to acquire unique integers from.
 // the acquired integers are used from the network service for defining the:
+// one integer for:
 // - vrf name
 // - vni
 // - vxlan-id
+// and one integer for:
+// - asn-id offset added to 4210000000 (ASNBase)
 //
 // the integer range has a vxlan-id constraint from Cumulus:
 // 	net add vxlan vxlan10 vxlan id
 //  <1-16777214>  :  An integer from 1 to 16777214
 //
+// FIXME Gerrit:
+// Aus dem Pool zieht man aber nicht die 4.000.000.000 raus.
+// Ist halt theoretisch möglich wie bei dem VRF 104009.
+// Man möchte 104010 einführen in seiner Infrastruktur und möglicherweise hat eine ASN dann diese ID herausgezogen und kein VRF. Weiß nicht, ob das so schlimm ist.^^
```

Comment:
This comment is partially superfluous with the seperation of pools, I guess?

To avoid conflicting VRF IDs with our well known ones (104009 and 104010), we could loop over the non private networks and acquire those IDs from the pool?

Reply:
Will remove this comment regarding seperation.
avoiding that these "private" vrf ids are acquired is not possible as long as the integerpool does not support such protected areas.

Reply:
after discussion with @mwindower there is no hard reason to reserve special vrf ids for private vrfs.